notes_app/notes/test_user_resource.py

```diff
@@ -0,0 +1,225 @@
+from django.contrib.auth.models import User
+from django.test import TestCase
+from tastypie.test import ResourceTestCaseMixin
+
+
+class UserResourceTest(ResourceTestCaseMixin, TestCase):
+    fixtures = ['test_users.json']
+
+    def setUp(self):
+        super(UserResourceTest, self).setUp()
+
+        self.username = 'admin'
+        self.password = 'admin13'
+        self.user = User.objects.get(username=self.username)
+
+        self.api_list = '/api/v1/user/'
+        self.api_detail = '/api/v1/user/1/'
+        self.api_user_id = "1"
+        self.post_data = {
+            "email": "test@gmail.com",
+            "first_name": "test",
+            "is_active": "true",
+            "is_staff": "false",
+            "is_superuser": "false",
+            "last_name": "test",
+            "password": "test",
+            "username": "test"
+        }
+        self.put_data = {
+            "first_name": "testPut"
+        }
+        self.patch_data = {
+            "last_name": "testPatch"
+        }
+
+    def get_credentials(self):
+        return self.create_basic(
+            username=self.user.username,
+            password=self.password)
+
+    def get_wrong_credentials(self):
+        return self.create_basic(
+            username=self.user.username,
+            password='111'
+        )
+
+    # Test List
+    def test_get_list_json(self):
+        """Authenticated User should get back a list of users."""
+        resp = self.api_client.get(
+            self.api_list,
+            format='json',
+            authentication=self.get_credentials()
+        )
+        total_count = User.objects.all().count()
+        self.assertValidJSONResponse(resp)
+        self.assertEqual(len(self.deserialize(resp)['objects']), total_count)
+        self.assertHttpOK(resp)
+
+    def test_post_list_and_delete_detail(self):
+        """Authorized Users can POST list and DELETE detail"""
+        users_before = User.objects.all().count()
+
+        # create
+        self.assertHttpCreated(
+            self.api_client.post(
+                self.api_list,
+                format='json',
+                data=self.post_data,
+                authentication=self.get_credentials()
+            )
+        )
+        users_after = User.objects.all().count()
+        self.assertEqual((users_before + 1), users_after)
+
+        # delete
+        new_user_uri = self.deserialize(
+            self.api_client.get(
+                '/api/v1/user/{0}/'.format(users_after),
+                format='json',
+                authentication=self.get_credentials()
+            )
+        )[u'resource_uri']
```

Comment:
I think it's easier to import `unicode_literals` one time and use all strings as unicode http://python-future.org/unicode_literals.html